### PR TITLE
requestAnimationFrame RFC

### PR DESCRIPTION
**Pull request by dakom**
initial draft of the rfc for requestAnimationFrame API

### CODE REVIEW

**Comment by dakom** [comment]
Should I add `RafState` into the PR?

**Comment by dakom** [comment]
added it in :)

File: rfcs/002-mid-level-raf-api.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+}
+
+```
+
+## Prior Art
+
+The initial commit of this RFC uses the code verbatim from [dominator](https://github.com/Pauan/rust-dominator/) (with permission from the author)
+
+The wasm-bindgen guide proposes [another approach](https://rustwasm.github.io/wasm-bindgen/examples/request-animation-frame.html) which is similar in spirit but less gloo-like. Specifically, adapting this into a generic API that supports cancellation would likely require that the user call a `cancel()` function explicitly, rather than the RAII style proposed here.
```
Comment (yoshuawuyts) [inline rfcs/002-mid-level-raf-api.md]: I'm unsure about the meaning of "gloo-like" here; I don't feel like we quite have concensus on what "gloo" APIs should be like, so perhaps leave this part out? The next sentence feels clear in explaining what the differences are.

Comment (dakom) [reply rfcs/002-mid-level-raf-api.md]: good catch - just pushed a change with that phrase removed

**Pull request merged**